pandas/core/missing.py

```diff
@@ -169,7 +169,9 @@ def _interp_limit(invalid, fw_limit, bw_limit):
     # the beginning (see issues #9218 and #10420)
     violate_limit = sorted(start_nans)
 
-    if limit:
+    if limit is not None:
+        if is_integer(limit) and limit < 1:
```

Comment:
actually I think we should make this a stronger check. ``limit`` MUST be an integer and > 0 if its not None. This would need checking both in cython and here.